src/tool_writeout_json.c

```diff
@@ -0,0 +1,223 @@
+  {"url_effective", JSON_STRING, CURLINFO_EFFECTIVE_URL},
+  {"http_code", JSON_LONG, CURLINFO_RESPONSE_CODE},
+  {"response_code", JSON_LONG, CURLINFO_RESPONSE_CODE},
+  {"http_connect", JSON_LONG, CURLINFO_HTTP_CONNECTCODE},
+  {"time_total", JSON_TIME, CURLINFO_TOTAL_TIME_T},
+  {"time_namelookup", JSON_TIME, CURLINFO_NAMELOOKUP_TIME_T},
+  {"time_connect", JSON_TIME, CURLINFO_CONNECT_TIME_T},
+  {"time_appconnect", JSON_TIME, CURLINFO_APPCONNECT_TIME_T},
+  {"time_pretransfer", JSON_TIME, CURLINFO_PRETRANSFER_TIME_T},
+  {"time_starttransfer", JSON_TIME, CURLINFO_STARTTRANSFER_TIME_T},
+  {"size_header", JSON_LONG, CURLINFO_HEADER_SIZE},
+  {"size_request", JSON_LONG, CURLINFO_REQUEST_SIZE},
+  {"size_download", JSON_LONG, CURLINFO_SIZE_DOWNLOAD_T},
+  {"size_upload", JSON_LONG, CURLINFO_SIZE_UPLOAD_T},
+  {"speed_download", JSON_TIME, CURLINFO_SPEED_DOWNLOAD_T},
+  {"speed_upload", JSON_TIME, CURLINFO_SPEED_UPLOAD_T},
+  {"content_type", JSON_STRING, CURLINFO_CONTENT_TYPE},
+  {"num_connects", JSON_LONG, CURLINFO_NUM_CONNECTS},
+  {"time_redirect", JSON_TIME, CURLINFO_REDIRECT_TIME_T},
+  {"num_redirects", JSON_LONG, CURLINFO_REDIRECT_COUNT},
+  {"ftp_entry_path", JSON_STRING, CURLINFO_FTP_ENTRY_PATH},
+  {"redirect_url", JSON_STRING, CURLINFO_REDIRECT_URL},
+  {"ssl_verify_result", JSON_LONG, CURLINFO_SSL_VERIFYRESULT},
+  {"proxy_ssl_verify_result", JSON_LONG, CURLINFO_PROXY_SSL_VERIFYRESULT},
+  {"filename_effective", JSON_FILENAME, CURLINFO_NONE},
+  {"remote_ip", JSON_STRING, CURLINFO_PRIMARY_IP},
+  {"remote_port", JSON_LONG, CURLINFO_PRIMARY_PORT},
+  {"local_ip", JSON_STRING, CURLINFO_LOCAL_IP},
+  {"local_port", JSON_LONG, CURLINFO_LOCAL_PORT},
+  {"http_version", JSON_VERSION, CURLINFO_HTTP_VERSION},
+  {"scheme", JSON_STRING, CURLINFO_SCHEME},
+  {NULL, JSON_NONE, CURLINFO_NONE}
+};
+
+static const char *http_version[] = {
+  "0",   /* CURL_HTTP_VERSION_NONE */
+  "1",   /* CURL_HTTP_VERSION_1_0 */
+  "1.1", /* CURL_HTTP_VERSION_1_1 */
+  "2"    /* CURL_HTTP_VERSION_2 */
+  "3"    /* CURL_HTTP_VERSION_3 */
+};
+
+static void jsonEscape(FILE *stream, const char *in)
+{
+  const char *i = in;
+  const char *in_end = in + strlen(in);
+
+  for(; i < in_end; i++) {
+    switch(*i) {
+    case '\\':
+      fputs("\\\\", stream);
+      break;
+    case '\"':
+      fputs("\\\"", stream);
+      break;
+    case '\b':
+      fputs("\\b", stream);
+      break;
+    case '\f':
+      fputs("\\f", stream);
+      break;
+    case '\n':
+      fputs("\\n", stream);
+      break;
+    case '\r':
+      fputs("\\r", stream);
+      break;
+    case '\t':
+      fputs("\\t", stream);
+      break;
+    default:
+      if (*i < 32) {
+        fprintf(stream, "u%04x", *i);
+      }
+      else {
+        fputc(*i, stream);
+      }
+      break;
+    };
+  }
+}
+
+static int writeTime(FILE *str, CURL *curl, const char *key, CURLINFO ci)
+{
+  long val = 0;
+  if(CURLE_OK == curl_easy_getinfo(curl, ci, &val)) {
```

Comment:
The info options ending with `_T` get `curl_off_t` values returned, not longs. That's an important distinction on some platforms.